sql/catalyst/src/main/scala/org/apache/spark/sql/sources/aggregates.scala

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.sources
+
+import org.apache.spark.sql.types.DataType
+
+// Aggregate Functions in SQL statement.
+// e.g. SELECT COUNT(EmployeeID), Max(salary), deptID FROM dept GROUP BY deptID
+// aggregateExpressions are (COUNT(EmployeeID), Max(salary)), groupByColumns are (deptID)
+case class Aggregation(aggregateExpressions: Seq[Seq[AggregateFunc]],
```

Comment:
why it's Seq of Seq?

Reply:
If we want to support `Avg` later, `Avg` will be rewritten to `Sum`/`Count`. I am grouping all the aggregates  in this pushed down Aggregation class. For example,  `SELECT Max(c1), Sum(c1), Min(c2), Avg(c3) FROM test`, The pushed down Aggregation has `[[Max], [Sum], [Min], [Sum, Count]]`, so it's a Seq of Seq

Reply:
Rewriting avg to sum / count should be Spark's work and transparent to the underlying data source. I don't think Seq of Seq is necessary here.

Reply:
I will change back to `Seq[AggregateFunc]`